diff -uNr wxlua/wxLua/modules/wxbind/src/wxcore_bind.cpp wxlua_patched/wxLua/modules/wxbind/src/wxcore_bind.cpp
--- wxlua/wxLua/modules/wxbind/src/wxcore_bind.cpp	2017-08-16 17:50:18.000000000 +0200
+++ wxlua_patched/wxLua/modules/wxbind/src/wxcore_bind.cpp	2017-08-20 19:59:33.588994551 +0200
@@ -63,20 +63,6 @@
 wxCursor* wxLua_wxHOURGLASS_CURSOR = NULL;
 wxCursor* wxLua_wxCROSS_CURSOR = NULL;
 
-#if defined(__MINGW32__) || defined(__GNUWIN32__)
-    // FIX: "internal compiler error: output_operand: invalid expression as operand"
-    // We set their value again later since some compilers (gcc) won't set their value yet
-    static wxPoint wxLua_wxDefaultPosition = wxDefaultPosition;
-    static wxSize wxLua_wxDefaultSize      = wxDefaultSize;
-    static wxEventType wxLua_wxEVT_COMMAND_DIRPICKER_CHANGED  = wxEVT_COMMAND_DIRPICKER_CHANGED;
-    static wxEventType wxLua_wxEVT_COMMAND_FILEPICKER_CHANGED = wxEVT_COMMAND_FILEPICKER_CHANGED;
-
-    #define wxDefaultPosition wxLua_wxDefaultPosition
-    #define wxDefaultSize     wxLua_wxDefaultSize
-    #define wxEVT_COMMAND_DIRPICKER_CHANGED  wxLua_wxEVT_COMMAND_DIRPICKER_CHANGED
-    #define wxEVT_COMMAND_FILEPICKER_CHANGED wxLua_wxEVT_COMMAND_FILEPICKER_CHANGED
-#endif //__MINGW32__
-
 
 // ---------------------------------------------------------------------------
 // wxLuaGetEventList_wxcore() is called to register events
@@ -7825,17 +7811,6 @@
 
 bool wxLuaBinding_wxcore::RegisterBinding(const wxLuaState& wxlState)
 {
-#if defined(__MINGW32__) || defined(__GNUWIN32__)
-    #undef wxDefaultPosition
-    #undef wxDefaultSize
-    #undef wxEVT_COMMAND_DIRPICKER_CHANGED
-    #undef wxEVT_COMMAND_FILEPICKER_CHANGED
-    wxLua_wxDefaultPosition = wxDefaultPosition;
-    wxLua_wxDefaultSize     = wxDefaultSize;
-    wxLua_wxEVT_COMMAND_DIRPICKER_CHANGED  = wxEVT_COMMAND_DIRPICKER_CHANGED;
-    wxLua_wxEVT_COMMAND_FILEPICKER_CHANGED = wxEVT_COMMAND_FILEPICKER_CHANGED;
-#endif //__MINGW32__
-
     // ugly hack for wxWidgets >2.7 change to use wxStockGDI::GetXXX
     if (wxLua_wxBLACK == NULL)
     {
