diff -uNr wxlua/wxLua/modules/wxbind/src/wxcore_bind.cpp wxlua_patched/wxLua/modules/wxbind/src/wxcore_bind.cpp
--- wxlua/wxLua/modules/wxbind/src/wxcore_bind.cpp	2017-08-16 17:50:18.000000000 +0200
+++ wxlua_patched/wxLua/modules/wxbind/src/wxcore_bind.cpp	2017-08-27 12:30:22.846836245 +0200
@@ -5652,7 +5652,7 @@
     extern wxLuaBindMethod wxCheckListBox_methods[];
     extern int wxCheckListBox_methodCount;
     static wxLuaArgType wxluabaseclass_wxluatypes_wxCheckListBox[] = { &wxluatype_wxItemContainer, &wxluatype_wxItemContainerImmutable, NULL };
-    static int wxluabaseclass_vtable_offsets_wxCheckListBox[] = { int(((long int)(wxItemContainer*)(wxCheckListBox*)&wxluatype_TNONE) - ((long int)(wxCheckListBox*)&wxluatype_TNONE)), int(((long int)(wxItemContainerImmutable*)(wxCheckListBox*)&wxluatype_TNONE) - ((long int)(wxCheckListBox*)&wxluatype_TNONE)) };
+    static int wxluabaseclass_vtable_offsets_wxCheckListBox[] = { int(((ptrdiff_t)(wxItemContainer*)(wxCheckListBox*)&wxluatype_TNONE) - ((ptrdiff_t)(wxCheckListBox*)&wxluatype_TNONE)), int(((ptrdiff_t)(wxItemContainerImmutable*)(wxCheckListBox*)&wxluatype_TNONE) - ((ptrdiff_t)(wxCheckListBox*)&wxluatype_TNONE)) };
     extern void wxLua_wxCheckListBox_delete_function(void** p);
 #endif // (wxLUA_USE_wxListBox && wxUSE_LISTBOX) && (wxLUA_USE_wxCheckListBox && wxUSE_CHECKLISTBOX)
 
@@ -5809,7 +5809,7 @@
 extern wxLuaBindMethod wxControlWithItems_methods[];
 extern int wxControlWithItems_methodCount;
 static wxLuaArgType wxluabaseclass_wxluatypes_wxControlWithItems[] = { &wxluatype_wxItemContainer, &wxluatype_wxItemContainerImmutable, NULL };
-static int wxluabaseclass_vtable_offsets_wxControlWithItems[] = { int(((long int)(wxItemContainer*)(wxControlWithItems*)&wxluatype_TNONE) - ((long int)(wxControlWithItems*)&wxluatype_TNONE)), int(((long int)(wxItemContainerImmutable*)(wxControlWithItems*)&wxluatype_TNONE) - ((long int)(wxControlWithItems*)&wxluatype_TNONE)) };
+static int wxluabaseclass_vtable_offsets_wxControlWithItems[] = { int(((ptrdiff_t)(wxItemContainer*)(wxControlWithItems*)&wxluatype_TNONE) - ((ptrdiff_t)(wxControlWithItems*)&wxluatype_TNONE)), int(((ptrdiff_t)(wxItemContainerImmutable*)(wxControlWithItems*)&wxluatype_TNONE) - ((ptrdiff_t)(wxControlWithItems*)&wxluatype_TNONE)) };
 extern void wxLua_wxControlWithItems_delete_function(void** p);
 extern wxLuaBindMethod wxDisplayChangedEvent_methods[];
 extern int wxDisplayChangedEvent_methodCount;
@@ -6099,7 +6099,7 @@
     extern wxLuaBindMethod wxChoice_methods[];
     extern int wxChoice_methodCount;
     static wxLuaArgType wxluabaseclass_wxluatypes_wxChoice[] = { &wxluatype_wxItemContainer, &wxluatype_wxItemContainerImmutable, NULL };
-    static int wxluabaseclass_vtable_offsets_wxChoice[] = { int(((long int)(wxItemContainer*)(wxChoice*)&wxluatype_TNONE) - ((long int)(wxChoice*)&wxluatype_TNONE)), int(((long int)(wxItemContainerImmutable*)(wxChoice*)&wxluatype_TNONE) - ((long int)(wxChoice*)&wxluatype_TNONE)) };
+    static int wxluabaseclass_vtable_offsets_wxChoice[] = { int(((ptrdiff_t)(wxItemContainer*)(wxChoice*)&wxluatype_TNONE) - ((ptrdiff_t)(wxChoice*)&wxluatype_TNONE)), int(((ptrdiff_t)(wxItemContainerImmutable*)(wxChoice*)&wxluatype_TNONE) - ((ptrdiff_t)(wxChoice*)&wxluatype_TNONE)) };
     extern void wxLua_wxChoice_delete_function(void** p);
 #endif // wxLUA_USE_wxChoice && wxUSE_CHOICE
 
@@ -6140,7 +6140,7 @@
     extern wxLuaBindMethod wxComboBox_methods[];
     extern int wxComboBox_methodCount;
     static wxLuaArgType wxluabaseclass_wxluatypes_wxComboBox[] = { &wxluatype_wxItemContainer, &wxluatype_wxItemContainerImmutable, NULL };
-    static int wxluabaseclass_vtable_offsets_wxComboBox[] = { int(((long int)(wxItemContainer*)(wxComboBox*)&wxluatype_TNONE) - ((long int)(wxComboBox*)&wxluatype_TNONE)), int(((long int)(wxItemContainerImmutable*)(wxComboBox*)&wxluatype_TNONE) - ((long int)(wxComboBox*)&wxluatype_TNONE)) };
+    static int wxluabaseclass_vtable_offsets_wxComboBox[] = { int(((ptrdiff_t)(wxItemContainer*)(wxComboBox*)&wxluatype_TNONE) - ((ptrdiff_t)(wxComboBox*)&wxluatype_TNONE)), int(((ptrdiff_t)(wxItemContainerImmutable*)(wxComboBox*)&wxluatype_TNONE) - ((ptrdiff_t)(wxComboBox*)&wxluatype_TNONE)) };
     extern void wxLua_wxComboBox_delete_function(void** p);
 #endif // wxLUA_USE_wxComboBox && wxUSE_COMBOBOX
 
@@ -6408,7 +6408,7 @@
     extern wxLuaBindMethod wxListBox_methods[];
     extern int wxListBox_methodCount;
     static wxLuaArgType wxluabaseclass_wxluatypes_wxListBox[] = { &wxluatype_wxItemContainer, &wxluatype_wxItemContainerImmutable, NULL };
-    static int wxluabaseclass_vtable_offsets_wxListBox[] = { int(((long int)(wxItemContainer*)(wxListBox*)&wxluatype_TNONE) - ((long int)(wxListBox*)&wxluatype_TNONE)), int(((long int)(wxItemContainerImmutable*)(wxListBox*)&wxluatype_TNONE) - ((long int)(wxListBox*)&wxluatype_TNONE)) };
+    static int wxluabaseclass_vtable_offsets_wxListBox[] = { int(((ptrdiff_t)(wxItemContainer*)(wxListBox*)&wxluatype_TNONE) - ((ptrdiff_t)(wxListBox*)&wxluatype_TNONE)), int(((ptrdiff_t)(wxItemContainerImmutable*)(wxListBox*)&wxluatype_TNONE) - ((ptrdiff_t)(wxListBox*)&wxluatype_TNONE)) };
     extern void wxLua_wxListBox_delete_function(void** p);
 #endif // wxLUA_USE_wxListBox && wxUSE_LISTBOX
 
@@ -6700,7 +6700,7 @@
     extern wxLuaBindMethod wxTextCtrl_methods[];
     extern int wxTextCtrl_methodCount;
     static wxLuaArgType wxluabaseclass_wxluatypes_wxTextCtrl[] = { &wxluatype_wxTextEntry, NULL };
-    static int wxluabaseclass_vtable_offsets_wxTextCtrl[] = { int(((long int)(wxTextEntry*)(wxTextCtrl*)&wxluatype_TNONE) - ((long int)(wxTextCtrl*)&wxluatype_TNONE)) };
+    static int wxluabaseclass_vtable_offsets_wxTextCtrl[] = { int(((ptrdiff_t)(wxTextEntry*)(wxTextCtrl*)&wxluatype_TNONE) - ((ptrdiff_t)(wxTextCtrl*)&wxluatype_TNONE)) };
     extern void wxLua_wxTextCtrl_delete_function(void** p);
     extern wxLuaBindMethod wxTextEntry_methods[];
     extern int wxTextEntry_methodCount;
diff -uNr wxlua/wxLua/modules/wxbind/src/wxhtml_bind.cpp wxlua_patched/wxLua/modules/wxbind/src/wxhtml_bind.cpp
--- wxlua/wxLua/modules/wxbind/src/wxhtml_bind.cpp	2017-08-16 17:50:18.000000000 +0200
+++ wxlua_patched/wxLua/modules/wxbind/src/wxhtml_bind.cpp	2017-08-27 12:31:42.495181751 +0200
@@ -5197,7 +5197,7 @@
     extern wxLuaBindMethod wxSimpleHtmlListBox_methods[];
     extern int wxSimpleHtmlListBox_methodCount;
     static wxLuaArgType wxluabaseclass_wxluatypes_wxSimpleHtmlListBox[] = { &wxluatype_wxHtmlWindowInterface, NULL };
-    static int wxluabaseclass_vtable_offsets_wxSimpleHtmlListBox[] = { int(((long int)(wxHtmlWindowInterface*)(wxSimpleHtmlListBox*)&wxluatype_TNONE) - ((long int)(wxSimpleHtmlListBox*)&wxluatype_TNONE)) };
+    static int wxluabaseclass_vtable_offsets_wxSimpleHtmlListBox[] = { int(((ptrdiff_t)(wxHtmlWindowInterface*)(wxSimpleHtmlListBox*)&wxluatype_TNONE) - ((ptrdiff_t)(wxSimpleHtmlListBox*)&wxluatype_TNONE)) };
     extern void wxLua_wxSimpleHtmlListBox_delete_function(void** p);
 #endif // (wxLUA_USE_wxHTML && wxUSE_HTML) && (wxCHECK_VERSION(2,8,0))
 
