diff -uNr wxlua/wxLua/modules/wxbind/src/wxcore_controls.cpp wxlua_patched/wxLua/modules/wxbind/src/wxcore_controls.cpp
--- wxlua/wxLua/modules/wxbind/src/wxcore_controls.cpp	2017-08-16 17:50:18.000000000 +0200
+++ wxlua_patched/wxLua/modules/wxbind/src/wxcore_controls.cpp	2017-08-27 12:52:41.679608993 +0200
@@ -1986,7 +1986,7 @@
 static int LUACALL wxLua_wxItemContainer_Append1(lua_State *L)
 {
     // voidptr_long number
-    long number = (long)wxlua_getnumbertype(L, 3);
+    ptrdiff_t number = (ptrdiff_t)wxlua_getnumbertype(L, 3);
     // const wxString item
     const wxString item = wxlua_getwxStringtype(L, 2);
     // get this
@@ -2074,7 +2074,7 @@
     // get this
     wxItemContainer * self = (wxItemContainer *)wxluaT_getuserdatatype(L, 1, wxluatype_wxItemContainer);
     // call GetClientData
-    long  returns = (long )self->GetClientData(n);
+    ptrdiff_t returns = (ptrdiff_t)self->GetClientData(n);
     // push the result number
     lua_pushnumber(L, returns);
 
@@ -2160,7 +2160,7 @@
 static int LUACALL wxLua_wxItemContainer_Insert1(lua_State *L)
 {
     // voidptr_long number
-    long number = (long)wxlua_getnumbertype(L, 4);
+    ptrdiff_t number = (ptrdiff_t)wxlua_getnumbertype(L, 4);
     // unsigned int pos
     unsigned int pos = (unsigned int)wxlua_getuintegertype(L, 3);
     // const wxString item
@@ -2202,7 +2202,7 @@
 static int LUACALL wxLua_wxItemContainer_SetClientData(lua_State *L)
 {
     // voidptr_long number
-    long number = (long)wxlua_getnumbertype(L, 3);
+    ptrdiff_t number = (ptrdiff_t)wxlua_getnumbertype(L, 3);
     // unsigned int n
     unsigned int n = (unsigned int)wxlua_getuintegertype(L, 2);
     // get this
@@ -5088,7 +5088,7 @@
     wxListCtrl *self = (wxListCtrl *)wxluaT_getuserdatatype(L, 1, wxluatype_wxListCtrl);
 
     // call SortItems
-    bool returns = self->SortItems(wxLua_ListCompareFunction, (long)&LCF_data);
+    bool returns = self->SortItems(wxLua_ListCompareFunction, (ptrdiff_t)&LCF_data);
 
     luaL_unref(L, LUA_REGISTRYINDEX, LCF_data.lua_tag); // remove ref to function
 
@@ -13548,7 +13548,7 @@
     // get this
     wxTreeItemId *self = (wxTreeItemId *)wxluaT_getuserdatatype(L, 1, wxluatype_wxTreeItemId);
     // call GetValue
-    long returns = (long)self->m_pItem;
+    ptrdiff_t returns = (ptrdiff_t)self->m_pItem;
     // push the result number
     lua_pushnumber(L, returns);
     // return the number of parameters
