diff -uNr wxlua/wxLua/modules/wxbind/src/wxbase_base.cpp wxlua_patched/wxLua/modules/wxbind/src/wxbase_base.cpp
--- wxlua/wxLua/modules/wxbind/src/wxbase_base.cpp	2017-08-16 17:50:18.000000000 +0200
+++ wxlua_patched/wxLua/modules/wxbind/src/wxbase_base.cpp	2017-08-27 13:12:47.459375861 +0200
@@ -3217,7 +3217,7 @@
     // get this
     wxEvtHandler * self = (wxEvtHandler *)wxluaT_getuserdatatype(L, 1, wxluatype_wxEvtHandler);
     // call GetClientData
-    long  returns = (long )self->GetClientData();
+    ptrdiff_t returns = (ptrdiff_t)self->GetClientData();
     // push the result number
     lua_pushnumber(L, returns);
 
@@ -3424,7 +3424,7 @@
 static int LUACALL wxLua_wxEvtHandler_SetClientData(lua_State *L)
 {
     // voidptr_long number
-    long number = (long)wxlua_getnumbertype(L, 2);
+    ptrdiff_t number = (ptrdiff_t)wxlua_getnumbertype(L, 2);
     // get this
     wxEvtHandler * self = (wxEvtHandler *)wxluaT_getuserdatatype(L, 1, wxluatype_wxEvtHandler);
     // call SetClientData
